WebEngine/JsonTransport.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "JsonTransport.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+
+#include "OrbitBase/Logging.h"
+
+namespace WebEngine {
+
+JsonTransport::JsonTransport(QWebSocket* socket) : socket_(socket) {
+  QObject::connect(socket_.get(), &QWebSocket::textMessageReceived, this,
+                   [this](const QString& msg) {
+                     QJsonParseError parse_error;
+                     const auto doc = QJsonDocument::fromJson(msg.toUtf8(), &parse_error);
+
+                     if (parse_error.error) {
+                       ERROR("Failed to parse web channel message: %s. The error was: %s",
```

Comment:
How long is a message typically? If it tends to span over multiple lines, maybe it would be better to have
"Failed to parse web channel message: \<error\>. Message:\n
\<message\>"
So that the error is on top and that the message starts on a new line.

Reply:
Yes, makes sense!

Reply:
Done.